pkg/resource/deploy/step_executor.go

```diff
@@ -172,8 +177,31 @@ func (se *stepExecutor) ExecuteParallel(antichain antichain) completionToken {
 
 // ExecuteRegisterResourceOutputs services a RegisterResourceOutputsEvent synchronously on the calling goroutine.
 func (se *stepExecutor) ExecuteRegisterResourceOutputs(e RegisterResourceOutputsEvent) error {
-	// Look up the final state in the pending registration list.
+	return se.executeRegisterResourceOutputs(e, false /* errored */, false /* finalizingStackOutputs */)
+}
+
+func (se *stepExecutor) executeRegisterResourceOutputs(
+	e RegisterResourceOutputsEvent,
+	errored,
+	finalizingStackOutputs bool,
```

Comment:
I feel that this parameter is violating SRP and causing this function to be responsible for parameter checking and causing side-effects that could be implemented more closely to the callsites that were added/modified in this PR:
- ExecuteRegisterResourceOutputs
- FinalizeStackOutputs

Reply:
I spent some time trying to rework things and didn't come up with anything much better. @Frassle, curious if you have any ideas on how this could be improved.